packages/oauth/src/createPublicOauthClient.ts

```diff
@@ -43,35 +43,85 @@ declare const process: {
   };
 };
 
+interface PublicOauthClientOptions {
+  client_id: string;
+  url: string;
+  redirectUrl: string;
+  useHistory?: boolean;
+  loginPage?: string;
+  postLoginPage?: string;
+  scopes?: string[];
+  fetchFn?: typeof globalThis.fetch;
+  ctxPath?: string;
+}
+
 /**
- * @param client_id
- * @param url the base url of your foundry server
- * @param redirectUrl the url configured for redirect in the oauth configuration on the server
- * @param useHistory if true, when possible uses `history.replaceState()`, otherwise uses `window.location.assign()`
- * @param loginPage if you want a custom landing page prior to logging in, set this to that url
- * @param postLoginPage if you want to customize where you return to after a completed authentication cycle. defaults to the url when the page loaded
- * @param scopes
- * @param fetchFn
- * @param ctxPath
- * @returns {PublicOauthClient} which can be used as a token provider
+ * Creates a PublicOauthClient for authentication.
+ *
+ * @param {string | PublicOauthClientOptions} clientIdOrOptions - Either the client_id as a string or an options object
+ * @param {string} [url] - The base URL of your Foundry server
+ * @param {string} [redirectUrl] - The URL configured for redirect in the OAuth configuration on the server
+ * @param {boolean} [useHistory=true] - If true, uses `history.replaceState()`, otherwise uses `window.location.assign()`
```

Comment:
Unfortunately JSDocs in VSCode doesn't seem to support this syntax for default values `[useHistory=true]`